src/payment/strategies/offsite-payment-strategy.ts

```diff
@@ -42,4 +42,11 @@ export default class OffsitePaymentStrategy extends PaymentStrategy {
 
         return super.finalize();
     }
+
+    private _shouldSubmitFullPayload(payment: OrderPaymentRequestBody): boolean {
```

Comment:
I think you missed it in my last comment, you also need to change `payment` to `payment?`. That way, you don't need to manually assert the type when you call the method (i.e.: `this._shouldSubmitFullPayload(payment as OrderPaymentRequestBody)`). I usually avoid using `as` as much as possible because you want the type checker to warn you. In this case, `payment` can be undefined, but the warning is suppressed by `as`. Fortunately, you have a guard statement inside the method so you won't have any runtime error. But somebody else can easily remove the guard statement and there won't be any warning from the type checker.